tests/integration/test_etcd.py

```diff
@@ -40,4 +40,26 @@ async def test_etcd_datastore(kubernetes_cluster: model.Model):
     status = json.loads(result.results["stdout"])
     assert status["ready"], "Cluster isn't ready"
     assert status["datastore"]["type"] == "external", "Not bootstrapped against etcd"
-    assert status["datastore"]["external-url"] == f"https://{etcd.public_address}:{etcd_port}"
+    assert status["datastore"]["servers"] == [f"https://{etcd.public_address}:{etcd_port}"]
+
+
+@pytest.mark.abort_on_fail
+async def test_update_etcd_cluster(kubernetes_cluster: model.Model):
+    """Test that adding etcd clusters are propagated to the k8s cluster."""
+    k8s: unit.Unit = kubernetes_cluster.applications["k8s"].units[0]
+    etcd = kubernetes_cluster.applications["etcd"]
+
+    await etcd.add_unit()
+    await etcd.add_unit()
```

Comment:
Make sure there are at least 3 etcd units.  If you use this technique, you can re-run the same test over and over without it adding lots of etcd units.

```suggestion
    count = 3 - len(etcd.units)
    if count > 0:
        await etcd.add_unit(count=count)
    await model.wait_for_idle(status="active")
```